src/prefect/_internal/concurrency/calls.py

```diff
@@ -339,6 +340,14 @@ def _run_sync(self):
             logger.debug("Finished call %r", self)  # noqa: F821
 
     async def _run_async(self, coro):
+        from prefect._internal.concurrency.threads import in_global_loop
+
+        # Ensure the greenback portal is shimmed for this task so we can safely call
+        # back into async code from sync code if the user does something silly; avoid
+        # this overhead for our internal event loop
+        if not in_global_loop():
+            await greenback.ensure_portal()
+
```

Comment:
I had a tough time figuring out just _where_ to inject `ensure_portal`.

This is the only place that works without having the user do it themselves.

If we call this in the global event loop we actually can cause deadlocks.